### ISSUE TITLE
Training doesn't converge

### ISSUE THREAD

**Issue by boeboska** [nianticlabs/monodepth2#300]
Hello Clément,

I am investigating if training from scratch results in the same depth output as your pre trained models. Unfortunately this is not the case so far. I’m trying to figure out what I am doing wrong. In your paper I red that after 20 epochs training should be converged. 

I now have trained the model for 10 epochs. There is no change in the loss over time whereas I should expect at decrease in the loss over time. Is this loss behavior correct?

Also, I tested the model at inference after 3, 4 … 9 epochs. There seems to me no real difference between the depth images after different epoch times.

I changed the following settings for training om my pc for stable running:
Batch-size = 1
Num workers = 0

Do you have an idea what is going wrong?  I have added some images for clarification below.

Thanks for your help!
-	Bob Borsboom (Msc Artificial Intelligence student Amsterdam)

![image](https://user-images.githubusercontent.com/31690823/103535094-72032d00-4e90-11eb-8d1d-df7c91454f5c.png)

![image](https://user-images.githubusercontent.com/31690823/103535104-74fe1d80-4e90-11eb-90cc-0c90e144e7a8.png)

Test the model at inference after different times during training:
![image](https://user-images.githubusercontent.com/31690823/103535108-76c7e100-4e90-11eb-82fa-5ef28d08dc85.png)
Result from pre trained model:
![image](https://user-images.githubusercontent.com/31690823/103535117-7af3fe80-4e90-11eb-8fba-c707e66f409d.png)

Sample from training output:
![image](https://user-images.githubusercontent.com/31690823/103535140-83e4d000-4e90-11eb-89da-2115142d50be.png)

**Comment by vineeth2309** [comment]
Hi,
I am currently also facing a similar issue where the training does not converge when training with the mono+stereo approach. I have collected my own stereo dataset, and during training the loss fluctuates in the same region of around 0.16 even after training for 50 epochs. Also the predicted depths that i calculate using the pretrained model, and the model I train are very different. For the model I trained, it does not predict depths greater than 5m, compared to around 40m for the pretrained model.

**Comment by boeboska** [comment]
Hi,


My problem is solved now! I accidentally changed something in the code.

Thanks for the nice work!


Bob

**Comment by mattbev** [comment]
> Hi,
> 
> My problem is solved now! I accidentally changed something in the code.
> 
> Thanks for the nice work!
> 
> Bob

@boeboska what was the code snippet you corrected? I'm seeing the same issue when finetuning on my custom dataset

**Comment by CaptainEven** [comment]
@boeboska Hi, how you solved the problem?

**Comment by jamesheatonrdm** [comment]
@boeboska in future when you mention that you have changed 'something' it is standard practice to tell us what you have changed. As there are multiple people who are facing the same issue as you
